docs/dqx/docs/reference/quality_rules.mdx

```diff
@@ -470,6 +470,30 @@ checks = [
 ```
 </details>
 
+### Usage examples of check functions with Map and Array columns
+
+You can apply checks to complex columns (e.g. `MapType` or `ArrayType`) by passing column expressions to the check function:
+<details style={{ backgroundColor: 'transparent', color: 'neutral' }}>
+<summary>**Checks defined on complex columns**</summary>
+```python
+from pyspark.sql.functions import col, try_element_at
+from databricks.labs.dqx.rule import DQRule
+from databricks.labs.dqx.col_functions import *
+from datetime import datetime
+
+checks = [
+    DQRule(
+        criticality="error",
+        check=is_not_null(col("map_col").getItem("map_key"))
+    ),
+    DQRule(
+        criticality="error",
+        check=is_not_null(try_element_at("array_col", 1))
+    )
+]
+```
+</details>
```

Comment:
Please also provide the same examples but defined as yaml. We should always provide examples with classes and yaml.